index 41d2715d..4ae8f538 100644
===================================================================
--- usonic.orig/sm/sonic-swss/portsyncd/linksync.cpp
+++ usonic/sm/sonic-swss/portsyncd/linksync.cpp
@@ -205,7 +205,7 @@ void LinkSync::onMsg(int nlmsg_type, struct nl_object *obj)
      * coming from old interfaces.
      */
 
-    if (m_ifindexOldNameMap.find(ifindex) != m_ifindexOldNameMap.end())
+    if (nlmsg_type == RTM_DELLINK && m_ifindexOldNameMap.find(ifindex) != m_ifindexOldNameMap.end())
     {
         SWSS_LOG_INFO("nlmsg type:%d Ignoring message for old interface %s(%d)",
                 nlmsg_type, key.c_str(), ifindex);
